src/Data/Fin/Subset/Properties.agda

```diff
@@ -203,6 +256,171 @@ p⊆q⇒∣p∣<∣q∣ {p = outside ∷ p} {inside  ∷ q} p⊆q = ≤-step (p
+⊂′⇒⊂ {x = outside ∷ p} {inside  ∷ q} (op⊆iq , iq⊈op) = op⊆iq , zero , here , λ ()
+⊂′⇒⊂ {x = inside  ∷ p} {outside ∷ q} (ip⊆oq , oq⊈ip) = contradiction (ip⊆oq here) λ ()
+⊂′⇒⊂ {x = inside  ∷ p} {inside  ∷ q} (ip⊆iq , iq⊈ip) with ⊂′⇒⊂ (drop-∷-⊆ ip⊆iq , λ q⊆p → iq⊈ip (in⊆in q⊆p))
+⊂′⇒⊂ {x = inside  ∷ p} {inside  ∷ q} (ip⊆iq , iq⊈ip) | _ , x , x∈q , x∉p = ip⊆iq , suc x , there x∈q , λ {(there x∈p) → contradiction x∈p x∉p}
+
+⊂′-trans : ∀ {n} → Transitive (_⊂′_ {n})
+⊂′-trans p⊂′q q⊂′r = ⊂⇒⊂′ (⊂-trans (⊂′⇒⊂ p⊂′q) (⊂′⇒⊂ q⊂′r))
+
+⊂′-trans-⊆ : ∀ {n} → Trans (_⊂′_ {n}) (_⊆_ {n}) (_⊂′_ {n})
+⊂′-trans-⊆ p⊂′q q⊆r = ⊂⇒⊂′ (⊂-trans-⊆ (⊂′⇒⊂ p⊂′q) q⊆r)
+
+⊆-trans-⊂′ : ∀ {n} → Trans (_⊆_ {n}) (_⊂′_ {n}) (_⊂′_ {n})
+⊆-trans-⊂′ p⊆q q⊂′r = ⊂⇒⊂′ (⊆-trans-⊂ p⊆q (⊂′⇒⊂ q⊂′r))
+
+⊂′-irref : ∀ {n} → Irreflexive _≡_ (_⊂′_ {n})
+⊂′-irref refl (p⊂p , p⊈p) = p⊈p p⊂p
+
+⊂′-antisym : ∀ {n} → Antisymmetric _≡_ (_⊂′_ {n})
+⊂′-antisym (p⊆q , _) (q⊆p , _) = ⊆-antisym p⊆q q⊆p
+
+⊂′-asymmetric : ∀ {n} → Asymmetric (_⊂′_ {n})
+⊂′-asymmetric (p⊆q , _) (_ , p⊈q) = case p⊈q p⊆q of λ ()
+
+infix 4 _⊂′?_
+
+_⊂′?_ : ∀ {n} → B.Decidable (_⊂′_ {n = n})
+[]          ⊂′? []          = no (λ {([]⊆[] , []⊈[]) → []⊈[] []⊆[]})
+outside ∷ p ⊂′? outside ∷ q = Dec.map out⊂′out-⇔ (p ⊂′? q)
+outside ∷ p ⊂′? inside  ∷ q = Dec.map out⊂′in-⇔ (p ⊆? q)
+inside  ∷ p ⊂′? outside ∷ q = no (λ {(ip⊆oq , _) → contradiction (ip⊆oq here) λ ()})
+inside  ∷ p ⊂′? inside  ∷ q = Dec.map in⊂′in-⇔ (p ⊂′? q)
+
+module _ (n : ℕ) where
+
+  ⊂′-isStrictPartialOrder : IsStrictPartialOrder _≡_ (_⊂′_ {n})
+  ⊂′-isStrictPartialOrder = record
+    { isEquivalence = isEquivalence
+    ; irrefl = ⊂′-irref
+    ; trans = ⊂′-trans
+    ; <-resp-≈ = (λ {refl → id}) , (λ {refl → id})
+    }
+
+  ⊂′-strictPartialOrder : StrictPartialOrder _ _ _
+  ⊂′-strictPartialOrder = record
+    { _≈_ = _≡_
+    ; _<_ = _⊂′_
+    ; isStrictPartialOrder = ⊂′-isStrictPartialOrder
+    }
+
+  ⊂′-isDecStrictPartialOrder : IsDecStrictPartialOrder _≡_ (_⊂′_ {n})
+  ⊂′-isDecStrictPartialOrder = record
+    { isStrictPartialOrder = ⊂′-isStrictPartialOrder
+    ; _≟_ = ≡-dec _≟_
+    ; _<?_ = _⊂′?_
+    }
+
+  ⊂′-decStrictPartialOrder : DecStrictPartialOrder _ _ _
+  ⊂′-decStrictPartialOrder = record
+    { _≈_ = _≡_
+    ; _<_ = _⊂′_
+    ; isDecStrictPartialOrder = ⊂′-isDecStrictPartialOrder
+    }
+
+p⊂′q⇒∣p∣<∣q∣ : ∀ {n} → {p q : Subset n} → p ⊂′ q → ∣ p ∣ < ∣ q ∣
+p⊂′q⇒∣p∣<∣q∣ {p = []} {[]} (p⊆q , q⊈p) = case q⊈p p⊆q of λ ()
+p⊂′q⇒∣p∣<∣q∣ {p = outside ∷ p} {outside ∷ q} op⊂′oq@(_ , _) = p⊂′q⇒∣p∣<∣q∣ (drop-∷-⊂′ op⊂′oq)
+p⊂′q⇒∣p∣<∣q∣ {p = outside ∷ p} {inside ∷ q} (op⊆iq , _) = s≤s (p⊆q⇒∣p∣<∣q∣ (drop-∷-⊆ op⊆iq))
+p⊂′q⇒∣p∣<∣q∣ {p = inside ∷ p} {outside ∷ q} (ip⊆oq , _) = contradiction (ip⊆oq here) (λ ())
+p⊂′q⇒∣p∣<∣q∣ {p = inside ∷ p} {inside ∷ q} ip⊂′iq@(ip⊆iq , _) = s≤s (p⊂′q⇒∣p∣<∣q∣ (drop-∷-⊂′ ip⊂′iq))
+
+------------------------------------------------------------------------
+-- ∁
+
+x∉∁s⇒x∈s : ∀ {n} → {x : Fin n} → {s : Subset n} → x ∉ ∁ s → x ∈ s
```

Comment:
I think these last two will be significantly easier to prove by first proving the contrapositive (i.e. `x∈s⇒x∉∁s` and `x∈∁s⇒x∉s`) by induction.

Reply:
I'm not sure how to get from there within intuitionistic logic though so I switched to induction for those functions.